File: NOVEMBER-2025/18-11-2025/p717.kt
class Solution {
    fun isOneBitCharacter(bits: IntArray): Boolean {
        var i = 0
        while (true) {
            val bit = bits[i]
            if (bit == 0) {
                i++
                if (i == bits.size) {
                    return true
                }
            } else {
                i += 2
                if (i >= bits.size) {
                    return false
                }
            }
        }
    }
}

File: NOVEMBER-2025/18-11-2025/p717.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut bits = bits.into_iter();
        let mut last_zero = false;
        while let Some(b) = bits.next() {
            if b == 0 {
                last_zero = true;
                continue;
            }
            last_zero = false;
            bits.next();
        }
        last_zero
    }
}